zkvm/docs/zkvm-spec.md

```diff
@@ -1201,6 +1202,23 @@ No changes to the [constraint system](#constraint-system) are made until [`verif
 
 Fails if `c1` and `c2` are not [constraints](#constraint-type).
 
+#### not
+
+_constr1_ **not** → _constr2_
+
+1. Pops [constraint](#constraint-type) `c1`.
+2. Create two constraints:
+   ```
+   x * y = 0
+   x * w = 1-y
+   ```
+   where `w` is a random challenge and `x` is the expression from constraint `c1`.
```

Comment:
`w` is not a random challenge, it's a free variable. 

Let's rephrase`x` to "`x` is the evaluation of the constraint `c1`". 

I'll add a section on evaluating expressions and constraints in the spec as a follow-up.